hubblestack/extmods/fdg/command_line_parser.py

```diff
@@ -0,0 +1,193 @@
+         chained (String): The command line in which to search for keys.
+                    This needs to be passed from another FDG module via chaining.
+         chained_status (boolean): Status returned by the chained method.
+Output (tuple):  tuple with two values. First item is the status of the module's execution.
+                Second item is list of values corresponding to the given key_aliases, fetched from the chained string.
+
+BASIC EXAMPLES:
+Inputs:
+    key_aliases = ["config-file"]
+    delimiter = "="
+    chained = "dockerd --config-file=/etc/docker/daemon.json --log-level=debug"
+Output: (True, ["/etc/docker/daemon.json"])
+
+UNIT TESTS:
+Check the unit test case file 'test_command_line_parser.py' to understand more examples on how this module works.
+
+NOTES:
+
+Understanding the regex(s):
+You can copy paste the regex(s) on this webpage - https://regex101.com/ to understand more about the regex.
+However, here is a small explanation
+
+regex_pattern :
+    "(?<=(\s|{))-{0,2}\"{0,1}\'{0,1}" + key_alias + "\"{0,1}\'{0,1}\s*" + delimiter + "\s*openingbracket.*closingbracket(?=(\s|$))"
+    This regex primarily focuses on whether the value is embedded inside brackets.
+
+1. '?<=' : denotes a positive lookbehind
+2. '\s' : matches the <space> character
+3. '{' : matches opening braces
+4. (\s|{) : () brackets signify a group. This is the first group in this regex
+4. -{0,2} : matches (zero to two) hyphens.
+5. \"{0,1}\'{0,1} : matches if any quotes are present.
+5. key_alias : matches the given key_alias.
+6. /s* : matches 0 or more <space> characters
+7. delimiter : matches the input delimiter
+8. openingbracket : is later replaced by one of the three opening bracket types
+9. .*? : matches any character lazily.
+10. closingbracket : is later replaced by one of the three closing bracket types
+11. (?=(\s|$|})) : matches that the end of the pattern must contain either <space> or EOL
+
+regex1 :
+    "(?<=(\s|{))-{0,2}\"{0,1}\'{0,1}" + key_alias + "\"{0,1}\'{0,1}\s*" + delimiter + "\s*([\",\']).*?\\2"
+    This regex primarily focuses on whether the value is embedded inside double or single quotes
+
+The starting regex is similar to the regex_pattern.
+1. ([\", \']) : () brackets signify a group. This is the second group in this regex.
+                    [\",\'] inside the () matches either a double-quote or a single-quote.
+2. .*? : matches any character lazily.
+3. \\2 : matches the second group of the regex.
+
+regex2 :
+    "(?<=(\s|{))-{0,2}\"{0,1}\'{0,1}" + key_alias + "\"{0,1}\'{0,1}\s*" + delimiter + "\s*.+?(?=(\s|$|}))"
+The starting regex is similar to the regex_pattern.
+
+1. (?=(\s|$)) : matches that the end of the pattern must contain either <space> or EOL
+"""
+
+import logging
+import re
+
+log = logging.getLogger(__name__)
+open_bracket_list = ["[","{","("]
+close_bracket_list = ["]","}",")"]
+
+
+def parse_cmdline(params='', chained=None, chained_status=None):
```

Comment:
Should use 'None' to represent null or nothing.

Reply:
Done.